### DISCUSSION TITLE
Time limited urls not expiring

### DISCUSSION THREAD

**Discussion by westor7** [YOURLS/YOURLS#3335] [Q&A] [answered]
Hello,

I set on config ``define('YOURLS_NONCE_LIFE', 300);`` at the bottom, but when i try to create a timed link from API, it creates that, but it doesn't expiring after `300` seconds, it keeps that link on the database and it is still working, i've followed https://docs.yourls.org/guide/advanced/passwordless-api.html#usage-of-a-time-limited-signature-token page and seems as the url is being created when calling it that seems that i succeed.

Using: 1.8.3

API Call Example: `https://go.domain.org/yourls-api.php?timestamp=1652025680&signature=31e049f4cef51fda31ef3dbab3b37ad5&hash=md5&action=shorturl&format=simple&url=https%3A%2F%2Fgoogle%2Ecom%2Ftest%5Fthis`

- Thanks!

**Comment by ozh** [comment, marked as answer]
I think you don't understand what this is for. The link you create via the API, eg http://sho.rt/blah, will be indefinitely valid, just like another short link created via the API or in the admin interface. What is time limited here is the API call, not its result.

If you want short links to be time limited, you'll need a plugin

**Reply by westor7** [reply]
Aha right, that makes sense, then i confused indeed.

I thought YOURLS had something for auto-expire by default, is it able to exist something like `&expire=1652025680` ? to remove the link from the database on the given time.

Or maybe something `&validfor=SECONDS` ?

- Thanks!

**Reply by ozh** [reply]
> If you want short links to be time limited, you'll need a plugin

⬆
